velox/functions/prestosql/DateTimeImpl.h

```diff
@@ -60,4 +63,124 @@ FOLLY_ALWAYS_INLINE std::optional<Timestamp> fromUnixtime(double unixtime) {
   auto nanos = unixtime - seconds;
   return Timestamp(seconds, nanos * kNanosecondsInSecond);
 }
+
+namespace {
+enum class DateTimeUnit {
+  kMillisecond,
+  kSecond,
+  kMinute,
+  kHour,
+  kDay,
+  kMonth,
+  kQuarter,
+  kYear
+};
+} // namespace
+
+// Year, quarter or month are not uniformly incremented. Months have different
+// total days, and leap years have more days than the rest. If the new year,
+// quarter or month has less total days than the given one, it will be coerced
+// to use the valid last day of the new month. For example,
```

Comment:
Please add :"use the valid last day of the new month. This could result in weird arithmetic behavior. For example,"